exporting/aws_kinesis/aws_kinesis.c

```diff
@@ -2,6 +2,44 @@
 
 #include "aws_kinesis.h"
 
+/**
+ * Clean AWS Kinesis *
+ */
+void aws_kinesis_cleanup(struct instance *instance)
+{
+    info("EXPORTING: cleaning up instance (aws_kinesis_cleanup) %s ...", instance->config.name);
+    if (instance->buffer)
+        buffer_free(instance->buffer);
+
+    if (instance->engine->aws_sdk_initialized)
+        aws_sdk_shutdown();
+
+    kinesis_shutdown(instance->connector_specific_data);
+
+    if (instance->connector_specific_data) {
```

Comment:
`free()` does nothing when a pointer is null, so the check is excessive. We also access all these pointers on Netdata startup - you'll never have an opportunity to get NULL pointer here, you'll get a segmentation fault much earlier.

Reply:
I also changed this